### DISCUSSION TITLE
Allow btcpayserver to assign a address permanently to a Data/user

### DISCUSSION THREAD

**Discussion by lucidddreams** [btcpayserver/btcpayserver#2071] [Ideas / Feature Requests]
**Is your feature request related to a problem? Please describe.**
The current btcpayserver implementation is limited to invoicing which cannot be applied for bitcoin exchange services or any alike

**Describe the solution you'd like**
Allow API to assign permanent bitcoin address to specific user or value. https://api.bitgo.com/docs/#operation/v2.wallet.addresses

**Sketch/Image/Wireframe/Mockup**
 If applicable provide examples, wireframes, sketches or images to better explain your idea.
n/a

**Describe alternatives you've considered**

**Additional context**
It would be great is btcpayserver will also have a bitgo.com alike feature where a master wallet can have multiple wallets under it which can be assigned to specific user or data

**Comment by NicolasDorier** [comment]
Those features are concerning more a pure bitcoin wallet. BTCPay Server isn't a Bitcoin wallet, it is a payment processor. The concept of "attaching an address to a user", doesn't really have any relation with invoices, orders or things relevant to payment processors.

As for the ability of hierarchical wallets, it doesn't make sense either: A store has a single wallet. What would it means if a store had several wallet? Which wallet to use when a new invoice is created? How would the UX change?
